gnovm/pkg/gnolang/preprocess.go

```diff
@@ -2510,37 +2983,39 @@ func isSwitchLabel(ns []Node, label Name) bool {
 }
 
 // Idempotent.
+// Also makes sure the stack doesn't reach MaxUint8 in length.
 func pushInitBlock(bn BlockNode, last *BlockNode, stack *[]BlockNode) {
 	if !bn.IsInitialized() {
-		bn.InitStaticBlock(bn, *last)
+		switch bn.(type) {
+		case *IfCaseStmt, *SwitchClauseStmt:
+			// skip faux block
+			bn.InitStaticBlock(bn, (*last).GetParentNode(nil))
+		default:
+			bn.InitStaticBlock(bn, *last)
+		}
 	}
 	if bn.GetStaticBlock().Source != bn {
 		panic("expected the source of a block node to be itself")
 	}
 	*last = bn
 	*stack = append(*stack, bn)
-}
-
-// like pushInitBlock(), but when the last block is a faux block,
-// namely after SwitchStmt and IfStmt.
-// Idempotent.
-func pushInitRealBlock(bn BlockNode, last *BlockNode, stack *[]BlockNode) {
-	if !bn.IsInitialized() {
-		bn.InitStaticBlock(bn, (*last).GetParentNode(nil))
-	}
-	if bn.GetStaticBlock().Source != bn {
-		panic("expected the source of a block node to be itself")
+	if len(*stack) >= math.MaxUint8 {
+		panic("block node depth reached maximum MaxUint8")
 	}
-	*last = bn
-	*stack = append(*stack, bn)
 }
 
 // like pushInitBlock(), but when the last block is a faux block,
 // namely after SwitchStmt and IfStmt.
 // Not idempotent, as it calls bn.Define with reference to last's TV value slot.
-func pushInitRealBlockAndCopy(bn BlockNode, last *BlockNode, stack *[]BlockNode) {
+// XXX do we need this?
```

Comment:
if this isn't used anywhere, we can remove the function. please check.

Reply:
this is needed, remove comment.